docs/limitations.md

```diff
@@ -0,0 +1,69 @@
+# Current Limitations and Installation Alternatives
+
+At least until [#3490](https://github.com/rectorphp/rector/issues/3490) is resolved, Rector relies on _runtime PHP reflection_ to analyze your source code. That means the code Rector works on cannot merely be treated as _data_, but has to be loaded (through the autoloader) into memory to be analyzed by the PHP interpreter.
+
+> Read the [PHPStan 0.12.26 release announcement](https://phpstan.org/blog/zero-config-analysis-with-static-reflection) to learn more about runtime and static reflection. PHPStan is the static analysis tool used by Rector.
+
+This brings a few limitations and affects your choices of how you can run Rector.
+
+## PHP Language Level
+
+The PHP process that runs Rector code also needs to load and parse your source code. So, you need to run a version of PHP that is both compatible with your project code as well as Rector.
+
+This means you cannot run Rector on PHP 8 to work on a project that uses language features limited to previous PHP versions.
+
+> One example for this is [array and string offset access using curly braces](https://www.php.net/manual/de/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace).
+
+But also the other way round, you cannot run Rector on PHP 7 to analyze a project with PHP 8 language features, like [constructor property promotion](https://stitcher.io/blog/constructor-promotion-in-php-8).
+
+Together with Rector's minimum PHP 7.3 version requirement, this might limit your options of how to run Rector.
+
+Note that using an older version of Rector might allow you to run a version of PHP <7.3.
+
+## Code Collisions
+
+Since Rector needs to load the code it works on in addition to the code it needs to run itself, conflicts may occur.
+
+One practical example of this is when you're running Rector on a project that uses an older version of Symfony. Since Rector uses a few Symfony Components itself, you may either run into problems when classes are re-declared, or when Rector happens to load and later (by accident) run parts of Symfony from your project installation. In the later case, the mix-up of different versions may lead to generally unpredictable results.
+
+To remedy this situation, [Rector Prefixed](https://github.com/rectorphp/rector-prefixed) has been created. Basically, it takes Rector and all of its dependencies and moves _all_ that code under a unique namespace prefix.
+
+(Add a note here: Drawbacks? Why isn't Rector-prefixed _the_ recommended approach?)
```

Comment:
Anyone?

Reply:
That's meta information users don't need. See PHPStan for example, the release produces such version.
It should be as part of Rector release too. Fully downgraded and prefixed version, e.g. under sub tag, 0.9.50.71

Reply:
I am afraid I cannot make sense of your comment in this context 😞. 

Is this referring to just why Rector-prefixed is not the primary recommended approach? (What are the drawbacks of using it?)

Or are you suggesting to remove the explanation what Rector-prefixed is about at all?

Reply:
> Or are you suggesting to remove the explanation what Rector-prefixed is about at all?

:+1: 

Unless Docker, the install command for Rector should be always the same:

```bash
composer install rector/rector
```

Composer itself should resolve if the version is prefixed (because of conflicts), or PHP 7.1 (because of lower PHP version). We should handle the release and tagging, instead of telling user the information mentioned above.

Reply:
Hm, I'll have to let that sink in a bit... 

I've tried such thing (offer several versions with "smart" tags/versions, let Composer find the "best" one) with internal packages in the past, and it turned out not to be possible in all cases, required a lot of effort to get it right and resulted in unintuitive version numbers (not what you'd expect from SemVer). Not to say it's not doable, just ...

But back to the initial question – what's the drawback of using `-prefixed`, at least for the average user? Is it just that it is less tested, does it bring other problems (for users, not developers)?